Create release notes for 2.0.3 (#6261)
PrefectHQ/prefect: RELEASE-NOTES.md

# Prefect Release Notes

## 2.0.3

This release contains a number of bug fixes and documentation improvements.

> ### Introducing `prefect-dbt`

Reviewer: Here's the link to the `prefect-dbt` docs:
```suggestion
### Introducing [`prefect-dbt`](https://prefecthq.github.io/prefect-dbt/)
```

> `prefect-dbt` is a collection of Prefect integrations for working with dbt with

Reviewer 2: ```suggestion
We've released `prefect-dbt` - a collection of Prefect integrations for working with dbt in your Prefect flows. This collection has been built as part of a partnership with dbt Labs to ensure that it follows best practices for working with dbt. 
```

> your Prefect flows.

Reviewer 2: ```suggestion
```

### Improvements and bug fixes

> - Users using Azure storage blocks can have ignored files.

Reviewer 2: ```suggestion
- Azure storage blocks can use `.prefectignore`
- Empty directories are no longer uploaded, preventing timeout errors with Azure
```

- Resolved bugs and improved interface in the Orion client. 
- Resolved a bug in Azure storage blocks that would cause uploads to get stuck.
- Corrected some erroneous type annotations.
- Better handling of database errors during state transition validation.
- Better sanitization of labels for Kubernetes Jobs.

> - Fixes `--manifest-only` flag of `deployment build` command.

Reviewer 2: ```suggestion
- Fixes `--manifest-only` flag of `deployment build` command to ensure that using this flag, the manifest gets generated, but the upload to a storage location is skipped.
```
